### PR TITLE
FIX: fallback to `anonymous` not `user` when suggesting usernames for anonymous users

### PR DESCRIPTION
**Pull request by AndrewPrigorshnev**
We use a translated string for generating usernames for anonymous users:
```ruby
username = I18n.t("anonymous").downcase
UserNameSuggester.suggest(username)
```

But the translation can contain only disallowed characters, in such cases we fall back to `User`. We want to be falling back to `anonymous` for anonymous users.

(based on @gschlager feedback [here](https://github.com/discourse/discourse/pull/14179#discussion_r698311871))

### CODE REVIEW

File: spec/services/anonymous_shadow_creator_spec.rb [hunk 1]
```diff
@@ -78,6 +78,16 @@
 
       expect { AnonymousShadowCreator.get(user) }.to_not raise_error
     end
-  end
 
+    it "fallback to username 'anonymous' if the translation for 'anonymous' consist from disallowed characters" do
```
Comment (eviltrout) [inline spec/services/anonymous_shadow_creator_spec.rb:82]:
```suggestion
    it "falls back to username 'anonymous' if the translation for 'anonymous' contains disallowed characters" do
```

**Review by eviltrout — APPROVED** [review]

File: spec/services/anonymous_shadow_creator_spec.rb (hunk 1 above)
Comment (AndrewPrigorshnev) [reply spec/services/anonymous_shadow_creator_spec.rb]: Would be "translation _consist of_ disallowed characters" correct? The idea here is that the translation contains _only_ disallowed characters. If only a part of characters is disallowed, we just strip them off.

Comment (eviltrout) [reply spec/services/anonymous_shadow_creator_spec.rb]: Good question. In that case I would say it should be "consists entirely of disallowed characters" or "consists solely of disallowed characters"

Comment (eviltrout) [reply spec/services/anonymous_shadow_creator_spec.rb]: (but to be honest I am not a stickler for language in test strings here. I'm pretty sure any dev will easily understand what you mean. It's just a suggestion!)

Comment (AndrewPrigorshnev) [reply spec/services/anonymous_shadow_creator_spec.rb]: "consists entirely of disallowed characters" sounds good, thank you!

File: app/services/anonymous_shadow_creator.rb [hunk 2]
```diff
@@ -76,4 +76,11 @@ def create_shadow!
       shadow
     end
   end
+
+  def resolve_username
+    username = I18n.t("anonymous").downcase
+    username = "anonymous" until UserNameSuggester.sanitize_username(username).present?
```
Comment (gschlager) [inline app/services/anonymous_shadow_creator.rb:82]: Huh! I've never seen `until` used that way. I had to think about this a couple of seconds...
I guess it works, it's just looks weird. :slightly_smiling_face:

I think the following should do the same and -- at least for me -- seems to be easier to understand. What do you think?

```suggestion
    username = UserNameSuggester.sanitize_username(username).presence || "anonymous"
```

Comment (AndrewPrigorshnev) [reply app/services/anonymous_shadow_creator.rb]: That was a silly typo, I meant `unless` not `until`. I've pushed a fix for that typo, @gschlager does the new version of the code make sense?

**Review by gschlager — APPROVED** [review]

**Pull request merged**